Sources/MixedReality/Microsoft.Psi.MixedReality.UniversalWindows/SpatialAnchorHelper.cs

```diff
@@ -56,6 +56,67 @@ public SpatialAnchor TryCreateSpatialAnchor(string id, SpatialCoordinateSystem s
 
             return spatialAnchor;
         }
+        
+        /// <summary>
+        /// Creates a persisted spatial anchor at the supplied <see cref="SpatialCoordinateSystem"/> with a positional offset.
+        /// </summary>
+        /// <param name="id">The identifier of the spatial anchor.</param>
+        /// <param name="spatialCoordinateSystem">The coordinate system at which to create the spatial anchor.</param>
+        /// <param name="translation">The rigid positional offset from the coordinate system's origin to the interop frame of reference's origin.</param>
+        /// <returns>The new spatial anchor, or null if the creation failed.</returns>
+        public SpatialAnchor TryCreateSpatialAnchor(string id, SpatialCoordinateSystem spatialCoordinateSystem, Vector3 translation)
+        {
+            // SpatialAnchor.TryCreateRelativeTo could return null if either the maximum number of
+            // spatial anchors has been reached, or if the world coordinate system could not be located.
+            var spatialAnchor = SpatialAnchor.TryCreateRelativeTo(spatialCoordinateSystem, translation);
+
+            if (spatialAnchor != null)
+            {
+                // Try to persist the spatial anchor to the store
+                if (this.spatialAnchorStore.TrySave(id, spatialAnchor))
+                {
+                    // Save it in the in-memory dictionary of spatial anchors
+                    this.spatialAnchors[id] = spatialAnchor;
+                }
+                else
+                {
+                    spatialAnchor = null;
+                }
+            }
+
+            return spatialAnchor;
+        }
+
+        /// <summary>
+        /// Creates a persisted spatial anchor at the supplied <see cref="SpatialCoordinateSystem"/> with offsets in position and rotation.
+        /// </summary>
+        /// <param name="id">The identifier of the spatial anchor.</param>
+        /// <param name="spatialCoordinateSystem">The coordinate system at which to create the spatial anchor.</param>
+        /// <param name="translation">The rigid positional offset from the coordinate system's origin to the interop frame of reference's origin.</param>
+        /// <param name="rotation">The rigid rotation from the coordinate system's origin to the interop frame of reference's origin.</param>
+        /// <returns>The new spatial anchor, or null if the creation failed.</returns>
+        public SpatialAnchor TryCreateSpatialAnchor(string id, SpatialCoordinateSystem spatialCoordinateSystem, Vector3 translation, System.Numerics.Quaternion rotation)
```

Comment:
>Vector3 translation, System.Numerics.Quaternion rotation [](http://example.com/codeflow?start=112&length=56)

What coordinate system basis is assumed in these parameters? It would be good to clarify that in the comments.

Would it be possible to instead pass in a `CoordinateSystem` in \psi-basis to specify the offset, and convert to HoloLens-basis vector and quaternion inside the method before creating the spatial anchor? Or maybe you would prefer to have the method like this, but then it would be nice to add one more method that does it like I described.

Reply:
From https://learn.microsoft.com/en-us/windows/mixed-reality/design/coordinate-systems#spatial-coordinate-systems:
> Spatial coordinate systems on Windows (and therefore, Windows Mixed Reality) are always right-handed.

I'd prefer to have a method like this, and I can make another one for the \psi coordinate system, no prob!

Reply:
Tested in both HL basis and \psi basis. Do you think the docs are clear enough in the last commit? Trying to convey that this allows users to specify a transform that is relative to the coordinate system generated by the spatial anchor.

Reply:
Thanks, this looks good. Maybe just add an additional comment in the remarks of this method and the one above that reminds about the definition of HoloLens basis (HoloLens basis means +x right, +y up, -z forward)